### PR TITLE
Fix `tanzu diagnostics collect` not respecting `--output-dir`

### LINKED ISSUE #1750: `tanzu diagnostics collect` not respecting `--output-dir` flag
## Bug Report

`tanzu diagnostics collect` command is not respecting the `--output-dir` flag. It always puts the tar ball out in the directory where the command is run

## Expected Behavior

`tanzu diagnostics collect` command to respect the `--output-dir` flag

## Steps to Reproduce the Bug

Try this with a kind cluster just for demo purposes

```bash
$ cd $HOME
$ kind create cluster --name kind
$ tanzu diagnostics collect --bootstrap-cluster-name kind --output-dir /tmp
$ ls /tmp/*.tar.gz
```

The tar ball won't be present in `/tmp` but it will instead be present in home directory

## Environment Details

* Build version (`tanzu version`): 

```bash
$ tanzu diagnostics version
v0.8.0-dev.3
```

Installed from https://storage.googleapis.com/tce-cli-plugins-staging/build-daily/127/tce-darwin-amd64-v0.8.0-dev.3.tar.gz

* Operating System (client): MacOS

## Additional information

The output directory flag variable is only used in the definition of the flag and to provide default value but not used anywhere else

default value - https://github.com/vmware-tanzu/community-edition/blob/ffafe9a8101c1ca8d68a9aecd40e8f74c3cbca52/cli/cmd/plugin/diagnostics/pkg/collect_cmd.go#L29

flag definition with default value included - https://github.com/vmware-tanzu/community-edition/blob/ffafe9a8101c1ca8d68a9aecd40e8f74c3cbca52/cli/cmd/plugin/diagnostics/pkg/collect_cmd.go#L67

---

The output directory variables are used in the starlark scripts too but are not used to put the tar ball file in that location. Below are the output directory variable usage in starlark scripts -

Bootstrap cluster script -

https://github.com/vmware-tanzu/community-edition/blob/ffafe9a8101c1ca8d68a9aecd40e8f74c3cbca52/cli/cmd/plugin/diagnostics/scripts/bootstrap_cluster.star#L71-L73

https://github.com/vmware-tanzu/community-edition/blob/ffafe9a8101c1ca8d68a9aecd40e8f74c3cbca52/cli/cmd/plugin/diagnostics/scripts/bootstrap_cluster.star#L86-L91

https://github.com/vmware-tanzu/community-edition/blob/ffafe9a8101c1ca8d68a9aecd40e8f74c3cbca52/cli/cmd/plugin/diagnostics/scripts/bootstrap_cluster.star#L5

But we don't use it for the tar ball file location -

https://github.com/vmware-tanzu/community-edition/blob/ffafe9a8101c1ca8d68a9aecd40e8f74c3cbca52/cli/cmd/plugin/diagnostics/scripts/bootstrap_cluster.star#L39

https://github.com/vmware-tanzu/community-edition/blob/ffafe9a8101c1ca8d68a9aecd40e8f74c3cbca52/cli/cmd/plugin/diagnostics/scripts/bootstrap_cluster.star#L41

Management cluster script -

https://github.com/vmware-tanzu/community-edition/blob/ffafe9a8101c1ca8d68a9aecd40e8f74c3cbca52/cli/cmd/plugin/diagnostics/scripts/management_cluster.star#L38-L40

https://github.com/vmware-tanzu/community-edition/blob/ffafe9a8101c1ca8d68a9aecd40e8f74c3cbca52/cli/cmd/plugin/diagnostics/scripts/management_cluster.star#L55-L62

https://github.com/vmware-tanzu/community-edition/blob/ffafe9a8101c1ca8d68a9aecd40e8f74c3cbca52/cli/cmd/plugin/diagnostics/scripts/management_cluster.star#L6

But we don't use it for the tar ball file location -

https://github.com/vmware-tanzu/community-edition/blob/ffafe9a8101c1ca8d68a9aecd40e8f74c3cbca52/cli/cmd/plugin/diagnostics/scripts/management_cluster.star#L24

https://github.com/vmware-tanzu/community-edition/blob/ffafe9a8101c1ca8d68a9aecd40e8f74c3cbca52/cli/cmd/plugin/diagnostics/scripts/management_cluster.star#L26

Workload cluster script -

https://github.com/vmware-tanzu/community-edition/blob/ffafe9a8101c1ca8d68a9aecd40e8f74c3cbca52/cli/cmd/plugin/diagnostics/scripts/workload_cluster.star#L43-L45

https://github.com/vmware-tanzu/community-edition/blob/ffafe9a8101c1ca8d68a9aecd40e8f74c3cbca52/cli/cmd/plugin/diagnostics/scripts/workload_cluster.star#L87-L95

https://github.com/vmware-tanzu/community-edition/blob/ffafe9a8101c1ca8d68a9aecd40e8f74c3cbca52/cli/cmd/plugin/diagnostics/scripts/workload_cluster.star#L6

But we don't use it for the tar ball file location -

https://github.com/vmware-tanzu/community-edition/blob/ffafe9a8101c1ca8d68a9aecd40e8f74c3cbca52/cli/cmd/plugin/diagnostics/scripts/workload_cluster.star#L29

https://github.com/vmware-tanzu/community-edition/blob/ffafe9a8101c1ca8d68a9aecd40e8f74c3cbca52/cli/cmd/plugin/diagnostics/scripts/workload_cluster.star#L31

An example of how including a directory name would look like - 

https://github.com/vmware-tanzu/crash-diagnostics/blob/3de8e0627581a11b32aec60fbd86aa39ae091284/examples/kind-api-objects.crsh#L21

---

Also as part of the golang code, we are not passing in the output directory variable value from the output directory flag value to the starlark scripts which have references to output directory variable as mentioned above. I think it has to be referenced in `argsMap` so that it's sent to the starlark script args as input

boostrap cluster starlark script args - https://github.com/vmware-tanzu/community-edition/blob/ffafe9a8101c1ca8d68a9aecd40e8f74c3cbca52/cli/cmd/plugin/diagnostics/pkg/collect_cmd.go#L127-L131

management cluster starlark script args - https://github.com/vmware-tanzu/community-edition/blob/ffafe9a8101c1ca8d68a9aecd40e8f74c3cbca52/cli/cmd/plugin/diagnostics/pkg/collect_cmd.go#L157-L162

workload cluster starlark script args - https://github.com/vmware-tanzu/community-edition/blob/ffafe9a8101c1ca8d68a9aecd40e8f74c3cbca52/cli/cmd/plugin/diagnostics/pkg/collect_cmd.go#L189-L198

### PR DESCRIPTION
**Pull request by karuppiah7890**
## What this PR does / why we need it

- Fixes golang module path of diagnostics plugin module
- Fixes `tanzu diagnostics collect` not respecting `--output-dir`
- Adds E2E tests

## Details for the Release Notes (PLEASE PROVIDE)
```release-note
Fix `tanzu diagnostics collect` not respecting `--output-dir`
```

## Which issue(s) this PR fixes

Fixes: #1750 

## Describe testing done for PR

Tested using the `e2e-test.sh` script in GitHub Actions using the workflow config present in this PR. Example run from my fork - https://github.com/karuppiah7890/community-edition/runs/3782918205?check_suite_focus=true

Tested locally too -

```bash
tce $ /Users/karuppiahn/projects/github.com/vmware-tanzu/tce/cli/cmd/plugin/diagnostics/e2e-test/e2e-test.sh 
Entering /Users/karuppiahn/projects/github.com/vmware-tanzu/tce/cli/cmd/plugin/diagnostics/e2e-test/.. directory to build tanzu diagnostics plugin
~/projects/github.com/vmware-tanzu/tce/cli/cmd/plugin/diagnostics ~/projects/github.com/vmware-tanzu/tce
Finished building tanzu diagnostics plugin. Leaving /Users/karuppiahn/projects/github.com/vmware-tanzu/tce/cli/cmd/plugin/diagnostics/e2e-test/..
~/projects/github.com/vmware-tanzu/tce
Creating a kind cluster for the E2E test
Creating cluster "e2e-diagnostics-181" ...
 ✓ Ensuring node image (kindest/node:v1.21.1) 🖼 
 ✓ Preparing nodes 📦  
 ✓ Writing configuration 📜 
 ✓ Starting control-plane 🕹️ 
 ✓ Installing CNI 🔌 
 ✓ Installing StorageClass 💾 
Set kubectl context to "kind-e2e-diagnostics-181"
You can now use your cluster with:

kubectl cluster-info --context kind-e2e-diagnostics-181

Have a nice day! 👋
Context "kind-e2e-diagnostics-181" renamed to "e2e-diagnostics-181-admin@e2e-diagnostics-181".
Running tanzu diagnostics collect command
2021/10/01 20:00:27 Collecting bootstrap cluster diagnostics
2021/10/01 20:00:27 Info: Bootstrap cluster: name=e2e-diagnostics-181
2021/10/01 20:00:27 Warn: Bootstrap cluster: specified name may not be valid: e2e-diagnostics-181
2021/10/01 20:00:27 Info: Found bootstrap cluster(s): ["e2e-diagnostics-181"]
2021/10/01 20:00:27 Info: Bootstrap cluster: e2e-diagnostics-181: capturing node logs
2021/10/01 20:00:30 Info: Capturing pod logs: cluster=e2e-diagnostics-181
2021/10/01 20:00:30 Info: Capturing API objects: cluster=e2e-diagnostics-181
2021/10/01 20:00:33 Info: Archiving: /var/folders/4z/09jpfvfj6c19lxl7ch78pzvc0000gn/T/tmp.27exp3N5/bootstrap.e2e-diagnostics-181.diagnostics.tar.gz
2021/10/01 20:00:33 Info: Capturing management cluster diagnostics: cluster=e2e-diagnostics-181; context=e2e-diagnostics-181-admin@e2e-diagnostics-181; kubeconfig=/Users/karuppiahn/.kube/config;
2021/10/01 20:00:33 Info: Capturing pod logs: cluster=e2e-diagnostics-181
2021/10/01 20:00:33 Info: Capturing API objects: cluster=e2e-diagnostics-181
2021/10/01 20:00:36 Info: Archiving: /var/folders/4z/09jpfvfj6c19lxl7ch78pzvc0000gn/T/tmp.27exp3N5/management-cluster.e2e-diagnostics-181.diagnostics.tar.gz
2021/10/01 20:00:36 Info: Retrieving workload cluster credentials
2021/10/01 20:00:36 Info: Retrieving workload cluster: cluster=e2e-diagnostics-181; context=e2e-diagnostics-181-admin@e2e-diagnostics-181; kubeconfig=/Users/karuppiahn/.kube/config;
2021/10/01 20:00:36 Info: Capturing pod logs: cluster=e2e-diagnostics-181
2021/10/01 20:00:36 Info: Capturing API objects: cluster=e2e-diagnostics-181
2021/10/01 20:00:40 Info: Archiving: /var/folders/4z/09jpfvfj6c19lxl7ch78pzvc0000gn/T/tmp.27exp3N5/workload-cluster.e2e-diagnostics-181.diagnostics.tar.gz
Checking if the diagnostics tar balls for the different clusters have been created
Cleaning up
Deleting cluster "e2e-diagnostics-181" ...
warning: this removed your active context, use "kubectl config use-context" to select a different one
deleted context e2e-diagnostics-181-admin@e2e-diagnostics-181 from /Users/karuppiahn/.kube/config
/var/folders/4z/09jpfvfj6c19lxl7ch78pzvc0000gn/T/tmp.27exp3N5/bootstrap.e2e-diagnostics-181.diagnostics.tar.gz
/var/folders/4z/09jpfvfj6c19lxl7ch78pzvc0000gn/T/tmp.27exp3N5/management-cluster.e2e-diagnostics-181.diagnostics.tar.gz
/var/folders/4z/09jpfvfj6c19lxl7ch78pzvc0000gn/T/tmp.27exp3N5/workload-cluster.e2e-diagnostics-181.diagnostics.tar.gz
/var/folders/4z/09jpfvfj6c19lxl7ch78pzvc0000gn/T/tmp.27exp3N5
tce $ 
```

## Special notes for your reviewer

Check the `e2e-test/README.md` for understanding what has been done in the E2E test script

### CODE REVIEW

**Review by vladimirvivien — COMMENTED** [review]
This is great work. Thanks for the additional testing code!!

File: cli/cmd/plugin/diagnostics/e2e-test/e2e-test.sh [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+TCE_VERSION="v0.9.1"
+
+echo "Installing TCE ${TCE_VERSION}"
+
+BUILD_OS=$(uname -s | tr '[:upper:]' '[:lower:]')
+TCE_RELEASE_TAR_BALL="tce-${BUILD_OS}-amd64-${TCE_VERSION}.tar.gz"
+TCE_RELEASE_DIR="tce-${BUILD_OS}-amd64-${TCE_VERSION}"
+INSTALLATION_DIR="${MY_DIR}/tce-installation"
+
+"${TCE_REPO_PATH}"/hack/get-tce-release.sh v0.9.1 "${BUILD_OS}"-amd64
```
[the remark this answers is not in the document]
Comment (karuppiah7890) [reply cli/cmd/plugin/diagnostics/e2e-test/e2e-test.sh:20]: Oops, yes. Good catch!

[the remark this answers is not in the document]
Comment (karuppiah7890) [reply cli/cmd/plugin/diagnostics/e2e-test/e2e-test.sh:20]: also, something to note is it currently supports only `amd64`. I didn't check more on checking the architecture from bash like `arm64`, 32 bit architectures etc

[the remark this answers is not in the document]
Comment (vladimirvivien) [reply cli/cmd/plugin/diagnostics/e2e-test/e2e-test.sh:20]: I think that's OK for now.

**Comment by karuppiah7890** [comment]
Looks like I have to close this PR unfortunately. The repo is detached from the PR 😅

**Comment by karuppiah7890** [comment]
I'll recreate the PR

File: cli/cmd/plugin/diagnostics/e2e-test/e2e-test.sh (hunk 1 above)
[the remark this answers is not in the document]
Comment (karuppiah7890) [reply cli/cmd/plugin/diagnostics/e2e-test/e2e-test.sh:20]: Fixed it in #2156 PR! 👍

**Pull request closed without merging**